environment/src/main/java/com/sequenceiq/environment/environment/flow/deletion/handler/EncryptionSetForResourceDeleteHandler.java

```diff
@@ -0,0 +1,76 @@
+import static com.sequenceiq.environment.environment.flow.deletion.event.EnvDeleteStateSelectors.START_PUBLICKEY_DELETE_EVENT;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.environment.environment.domain.Environment;
+import com.sequenceiq.environment.environment.dto.EnvironmentDeletionDto;
+import com.sequenceiq.environment.environment.dto.EnvironmentDto;
+import com.sequenceiq.environment.environment.flow.deletion.event.EnvDeleteEvent;
+import com.sequenceiq.environment.environment.flow.deletion.event.EnvDeleteFailedEvent;
+import com.sequenceiq.environment.environment.service.EnvironmentService;
+import com.sequenceiq.flow.reactor.api.event.EventSender;
+import com.sequenceiq.flow.reactor.api.handler.EventSenderAwareHandler;
+
+import reactor.bus.Event;
+
+@Component
+public class EncryptionSetForResourceDeleteHandler extends EventSenderAwareHandler<EnvironmentDeletionDto> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EncryptionSetForResourceDeleteHandler.class);
+
+    private final EnvironmentService environmentService;
+
+    protected EncryptionSetForResourceDeleteHandler(EventSender eventSender, EnvironmentService environmentService) {
+        super(eventSender);
+        this.environmentService = environmentService;
+    }
+
+    @Override
+    public String selector() {
+        return DELETE_ENCRYPTION_SET_FOR_RESOURCE_EVENT.selector();
+    }
+
+    @Override
+    public void accept(Event<EnvironmentDeletionDto> environmentDtoEvent) {
+        LOGGER.debug("Accepting EncryptionSetForResourceDelete event");
+        EnvironmentDeletionDto environmentDeletionDto = environmentDtoEvent.getData();
+        EnvironmentDto environmentDto = environmentDeletionDto.getEnvironmentDto();
+        EnvDeleteEvent envDeleteEvent = getEnvDeleteEvent(environmentDeletionDto);
+        try {
+            environmentService.findEnvironmentById(environmentDto.getId()).ifPresent(this::deleteEncryptionSet);
+            eventSender().sendEvent(envDeleteEvent, environmentDtoEvent.getHeaders());
+        } catch (Exception e) {
+            if (environmentDeletionDto.isForceDelete()) {
+                LOGGER.warn("The {} was not successful but the environment deletion was requested as force delete so " +
+                        "continue the deletion flow", selector());
+                eventSender().sendEvent(envDeleteEvent, environmentDtoEvent.getHeaders());
+            } else {
+                EnvDeleteFailedEvent failedEvent = new EnvDeleteFailedEvent(environmentDto.getId(),
+                        environmentDto.getName(), e, environmentDto.getResourceCrn());
+                eventSender().sendEvent(failedEvent, environmentDtoEvent.getHeaders());
+            }
+        }
+    }
+
+    private void deleteEncryptionSet(Environment environment) {
+        /*
+        TODO: Delete the Disk Encryption Set if it exists.
```

Comment:
And potentially update & save the environment. Although it is being deleted, so this may not be important. 😃